### PR TITLE
[release-1.9] 🌱 test: use a daemonset and kubectl exec to provide images to remote clusters

### PR DESCRIPTION
**Pull request by chrischdi**
Manual cherry-pick of https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2955

### CODE REVIEW

File: test/framework/image_preloading.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+			Stderr:    true,
+			TTY:       false,
+		}, scheme.ParameterCodec)
+
+	exec, err := remotecommand.NewSPDYExecutor(clusterProxy.GetRESTConfig(), "POST", req.URL())
+	if err != nil {
+		return err
+	}
+
+	var stdout, stderr bytes.Buffer
```
Comment (sbueringer) [inline test/framework/image_preloading.go:141]: Why is this missing the following?
```go
	// WebSocketExecutor must be "GET" method as described in RFC 6455 Sec. 4.1 (page 17).
	websocketExec, err := remotecommand.NewWebSocketExecutor(clusterProxy.GetRESTConfig(), "GET", req.URL().String())
	if err != nil {
		return err
	}
	exec, err = remotecommand.NewFallbackExecutor(websocketExec, exec, httpstream.IsUpgradeFailure)
	if err != nil {
		return err
	}
```

File: hack/e2e.sh [hunk 2]
```diff
@@ -144,16 +143,12 @@ echo "Acquired Workload Cluster Control Plane IP: $WORKLOAD_CONTROL_PLANE_ENDPOI
...
   mkdir -p /tmp/images
-  docker save gcr.io/k8s-staging-cluster-api/capv-manager:e2e -o "$DOCKER_IMAGE_TAR"
-
-  # store the image on gcs
-  login
-  E2E_IMAGE_SHA=$(docker inspect --format='{{index .Id}}' gcr.io/k8s-staging-cluster-api/capv-manager:e2e)
-  export E2E_IMAGE_SHA
-  gsutil cp ${DOCKER_IMAGE_TAR} gs://capv-ci/"$E2E_IMAGE_SHA"
+  docker save \
+    "gcr.io/k8s-staging-capi-vsphere/cluster-api-vsphere-controller-${ARCH}:dev" \
```
Comment (sbueringer) [inline hack/e2e.sh:150]: Not sure if the different image name works

File: test/framework/image_preloading.go (hunk 1 above)
Comment (chrischdi) [reply test/framework/image_preloading.go:141]: because these are not available in the used client-go version.

File: hack/e2e.sh (hunk 2 above)
Comment (chrischdi) [reply hack/e2e.sh]: good catch

File: test/framework/image_preloading.go (hunk 1 above)
Comment (sbueringer) [reply test/framework/image_preloading.go:141]: Okay, but it just works without?

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Pull request merged**